engine/bot-engine.js

```diff
@@ -156,7 +156,7 @@ let BotEngine = function(blocks, appContext) {
 
     async function fetchUserContext(c) {
         return await new Promise((resolve) => {
-            appContext.storage.getUserDataById(c.userId, c.messengerApi.messenger, (userData, newUser) => {
```

Comment:
This is questionable: I don't think the storage is not supposed to know about contexts. Let's discuss other options.